ivy.el

```diff
@@ -3210,7 +3239,16 @@ FACE is the face to apply to STR."
 (defun ivy--format-minibuffer-line (str)
   "Format line STR for use in minibuffer."
   (if (eq ivy-display-style 'fancy)
-      (funcall ivy--highlight-function (copy-sequence str))
+      (let ((str
+             (funcall ivy--highlight-function (copy-sequence str))))
+
+        (add-text-properties
```

Comment:
Why not `propertize`?

Reply:
Indeed, looks better with it, but adds an unnecessary copy. 

I guess that the `mouse-face` property can be added for both branches of the if, what do you think?

Reply:
> but adds an unnecessary copy

True, unless you pass the result of `propertize` to `ivy--highlight-function`:

```el
(defun ivy--format-minibuffer-line (str)
  "Format line STR for use in minibuffer."
  (if (eq ivy-display-style 'fancy)
      (funcall ivy--highlight-function
               (propertize str
                           'mouse-face 'ivy-minibuffer-match-highlight
                           'help-echo  '(format "mouse-1: %s"
                                                ivy-mouse-action-tooltip)))
    (copy-sequence str)))
```

Though this relies on `ivy--highlight-function` to not overwrite any of the `propertize` properties, so I don't support it too strongly.

Reply:
> I guess that the `mouse-face` property can be added for both branches of the `if`

Indeed, I think `mouse-face` should be set for any clickable results, whereas `ivy-display-style` seems to be more specific to highlighting regexp matches.